compiler/src/dotty/tools/dotc/core/OrderingConstraint.scala

```diff
@@ -217,6 +240,188 @@ class OrderingConstraint(private val boundsMap: ParamBounds,
       if tvar == null then NoType
       else tvar
 
+// ------------- Type parameter dependencies ----------------------------------------
+
+  private type ReverseDeps = SimpleIdentityMap[TypeParamRef, SimpleIdentitySet[TypeParamRef]]
+
+  /** A map that associates type parameters of this constraint with all other type
+   *  parameters that refer to them in their bounds covariantly, such that, if the
+   *  type parameter is instantiated to a larger type, the constraint would be narrowed
+   *  (i.e. solution set changes other than simply being made larger).
```

Comment:
I'm not sure I understand what this means. What is being made larger?

Reply:
The solution set. i.e. the constraint is relaxed.